src/main/java/com/eerussianguy/blazemap/feature/maps/AtlasExporter.java

```diff
@@ -0,0 +1,144 @@
+package com.eerussianguy.blazemap.feature.maps;
+
+import com.eerussianguy.blazemap.api.BlazeRegistry;
+import com.eerussianguy.blazemap.api.maps.Layer;
+import com.eerussianguy.blazemap.api.maps.MapType;
+import com.eerussianguy.blazemap.api.maps.TileResolution;
+import com.eerussianguy.blazemap.api.util.RegionPos;
+import com.eerussianguy.blazemap.engine.BlazeMapAsync;
+import com.eerussianguy.blazemap.engine.StorageAccess;
+import com.eerussianguy.blazemap.engine.client.BlazeMapClientEngine;
+import com.eerussianguy.blazemap.engine.client.LayerRegionTile;
+import com.eerussianguy.blazemap.util.Colors;
+import com.mojang.blaze3d.platform.NativeImage;
+import net.minecraft.client.Minecraft;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.world.level.Level;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AtlasExporter {
+    private static Task task = null;
+
+    public static synchronized void export(Task task) {
+        if(AtlasExporter.task == null) {
+            AtlasExporter.task = task;
+            BlazeMapAsync.instance().clientChain.runOnDataThread(AtlasExporter::export);
+        }
+    }
+
+    public static synchronized Task getTask() {
+        return task;
+    }
+
+    private static void export() {
+        TileResolution resolution = TileResolution.FULL;
+        StorageAccess.Internal storage = BlazeMapClientEngine.getDimensionStorage(task.dimension);
+        AtlasInfo atlasInfo = new AtlasInfo();
```

Comment:
Nit: Am not quite sure why you need a separate class to hold this data and can't include it either on the `AtlasExporter` or on the `Task`. Not a big deal though!

Reply:
It's true that it could be in Task.
This came from a multiple return point of view, where we explored using a class purely as a C struct that we pass in to be filled with the extras.
I think this whole feature needs some further clean up later, as there are other things like this that aren't problematic but need to be simplified.

Reply:
Sounds good. Defs not a blocker for just now

Reply:
Moved to `Task` in working branch